src/tokens/TokenBalanceItem.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View, ViewStyle } from 'react-native'
+import TokenDisplay from 'src/components/TokenDisplay'
+import TokenIcon from 'src/components/TokenIcon'
+import Touchable from 'src/components/Touchable'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { TokenBalance } from 'src/tokens/slice'
+
+interface Props {
+  token: TokenBalance
+  onPress?: () => void
+  containerStyle?: ViewStyle
+}
+
+export const TokenBalanceItem = ({ token, onPress, containerStyle }: Props) => {
+  const { t } = useTranslation()
+
+  const Content = ({ token, containerStyle }: Props) => {
+    return (
+      <View style={[styles.container, containerStyle]}>
+        <TokenIcon token={token} viewStyle={styles.marginRight} />
+        <View style={styles.textContainer}>
+          <View style={styles.line}>
+            <Text numberOfLines={1} style={[styles.label, styles.marginRight]}>
+              {token.name}
+            </Text>
+            <TokenDisplay
+              style={styles.amount}
+              amount={token.balance}
+              tokenId={token.tokenId}
+              showSymbol={true}
+              hideSign={true}
+              showLocalAmount={false}
+            />
+          </View>
+          <View style={styles.line}>
+            {token.networkId in NETWORK_NAMES ? (
+              <Text
+                numberOfLines={1}
+                style={[styles.subLabel, styles.marginRight]}
+                testID="NetworkLabel"
+              >
+                {t('assets.networkName', { networkName: NETWORK_NAMES[token.networkId] })}
+              </Text>
+            ) : (
+              <View />
+            )}
+            <TokenDisplay
+              style={styles.subAmount}
+              amount={token.balance}
+              tokenId={token.tokenId}
+              showSymbol={false}
+              hideSign={true}
+            />
+          </View>
+          {token.bridge && (
+            <Text
+              testID="BridgeLabel"
+              numberOfLines={1}
+              style={[styles.subLabel, { color: colors.informational }]}
+            >
+              {t('assets.bridge', { bridge: token.bridge })}
+            </Text>
+          )}
+        </View>
+      </View>
+    )
+  }
+
+  return typeof onPress !== undefined ? (
+    <Touchable onPress={onPress}>
+      <Content token={token} containerStyle={containerStyle} />
+    </Touchable>
+  ) : (
+    <Content token={token} containerStyle={containerStyle} />
+  )
+}
+
+const styles = StyleSheet.create({
+  container: {
+    marginHorizontal: Spacing.Thick24,
+    marginBottom: Spacing.Large32,
```

Comment:
Not super important since the styles are overridable, but could the marginBottom be split evenly to the top and the bottom? This would also allow us to re-use default styles on the Token Details screen and also the touchable area is split evenly instead of favoring one side.

Reply:
This might run into issues with vertical margin collapse where marginTop and marginBottom of adjacent items merge together, but I'll give it a try.

Reply:
Tested locally and it appears react native doesn't have this same issue that web does. Fixed in https://github.com/valora-inc/wallet/pull/4207/commits/465dff7375c3e506f1bffd0d6645276158755d19!